distribution/push_v2.go

```diff
@@ -360,11 +364,26 @@ func (pd *v2PushDescriptor) Upload(ctx context.Context, progressOutput progress.
 				return distribution.Descriptor{}, xfer.DoNotRetry{Err: err}
 			}
 			return err.Descriptor, nil
+		case errcode.Errors:
+			for _, e := range err {
+				switch e := e.(type) {
+				case errcode.Error:
+					if e.Code == errcode.ErrorCodeUnauthorized {
+						logrus.Debugf("error code is unauthoriedError")
+						isUnauthorizedError = true
+					}
+				default:
+				}
+			}
 		default:
 			logrus.Infof("failed to mount layer %s (%s) from %s: %v", diffID, mountCandidate.Digest, mountCandidate.SourceRepository, err)
 		}
 
-		if len(mountCandidate.SourceRepository) > 0 &&
+		if isUnauthorizedError && !pd.pushState.hasAuthInfo {
+			// break the attempt loop as user has not login
+			logrus.Infoln("failed to push image to registry because unauthorized error and user not login")
+			break
```

Comment:
I think this should continue instead of break. There might be other repos which do not require credentials to mount the blob. The continue can go inside the error checking code, instead of setting an `isUnauthorizedError` variable (it would mean having a label on the candidate loop and using `continue candidateLoop`).

Reply:
this is modified as you suggested